src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -22,11 +22,59 @@
 class YamlFileDumper extends FileDumper
 {
     /**
+     * Dump messages as tree, e.g. this
+     *   'foo.bar1' => 'test1',
+     *   'foo.bar2' => 'test2'
+     *
+     * dumps as follows:
+     *   foo:
+     *     bar1: test1
+     *     bar2: test2
+     *
+     * @var bool
+     */
+    private $asTree = true;
+
+    /**
+     * The level where you switch to inline YAML.
+     * Do not switch to inline YAML by default.
+     *
+     * @var int
+     */
+    private $inline = 999;
+
+    /**
+     * Sets asTree flag
+     * 
+     * @param bool $asTree
+     */
+    public function setDumpAsTree($asTree)
```

Comment:
The new abstract class makes the component more complex without any real benefit IMO.

And if you want to make the tree logic reusable, the right way would be to put it in its own class used as a collaborator of the dumper instead of using an abstract parent class. It would be much more reusable (and easier to test as you could test the tree logic on its own btw)